frontend/js/src/brainzplayer/AppleMusicPlayer.tsx

```diff
@@ -0,0 +1,365 @@
+/* eslint-disable no-underscore-dangle */
+import * as React from "react";
+import { debounce as _debounce, get as _get, isString } from "lodash";
+import { faApple } from "@fortawesome/free-brands-svg-icons";
+import { getArtistName, getTrackName, loadScriptAsync } from "../utils/utils";
+import { DataSourceProps, DataSourceType } from "./BrainzPlayer";
+
+export type AppleMusicPlayerProps = DataSourceProps & {
+  appleMusicUser?: AppleMusicUser;
+};
+
+export type AppleMusicPlayerState = {
+  currentAppleMusicTrack?: any;
+  progressMs: number;
+  durationMs: number;
+};
+
+export default class AppleMusicPlayer
+  extends React.Component<AppleMusicPlayerProps, AppleMusicPlayerState>
+  implements DataSourceType {
+  static hasPermissions = (appleMusicUser?: AppleMusicUser) => {
+    return true;
+  };
+
+  static isListenFromThisService = (listen: Listen | JSPFTrack): boolean => {
+    // Retro-compatibility: listening_from has been deprecated in favor of music_service
+    const listeningFrom = _get(
+      listen,
+      "track_metadata.additional_info.listening_from"
+    );
+    const musicService = _get(
+      listen,
+      "track_metadata.additional_info.music_service"
+    );
+    return (
+      (isString(listeningFrom) &&
+        listeningFrom.toLowerCase() === "apple_music") ||
+      (isString(musicService) &&
+        musicService.toLowerCase() === "music.apple.com") ||
+      Boolean(AppleMusicPlayer.getURLFromListen(listen))
+    );
+  };
+
+  static getURLFromListen(listen: Listen | JSPFTrack): string | undefined {
+    return _get(listen, "track_metadata.additional_info.apple_music_id");
+  }
+
+  public name = "Apple Music";
+  public domainName = "music.apple.com";
+  public icon = faApple;
+  // Saving the access token outside of React state , we do not need it for any rendering purposes
+  // and it simplifies some of the closure issues we've had with old tokens.
+  private accessToken = "";
+
+  private readonly _boundOnPlaybackStateChange: (event: any) => any;
+  private readonly _boundOnPlaybackTimeChange: (event: any) => any;
+  private readonly _boundOnPlaybackDurationChange: (event: any) => any;
+  private readonly _boundOnNowPlayingItemChange: (event: any) => any;
+
+  appleMusicPlayer?: AppleMusicPlayerType;
+  debouncedOnTrackEnd: () => void;
+
+  constructor(props: AppleMusicPlayerProps) {
+    super(props);
+    this.state = {
+      durationMs: 0,
+      progressMs: 0,
+    };
+
+    this.accessToken = props.appleMusicUser?.music_user_token || "";
+
+    this.debouncedOnTrackEnd = _debounce(props.onTrackEnd, 700, {
+      leading: true,
+      trailing: false,
+    });
+
+    // Do an initial check of the AppleMusic token permissions (scopes) before loading the SDK library
+    if (AppleMusicPlayer.hasPermissions(props.appleMusicUser)) {
+      window.addEventListener("musickitloaded", this.connectAppleMusicPlayer);
+      loadScriptAsync(
```

Comment:
I remember you telling that you had issues with installing new modules. Although using CDN is fine, but leads to security concerns and consistency issues (since we don't use this approach anywhere else)

so why don't you try
- installing musickit with npm, then 
- rebuilding `static_builder` with `--no-cache` flag